mavros_extras/src/plugins/onboard_computer_status.cpp

```diff
@@ -0,0 +1,121 @@
+ */
+class OnboardComputerStatusPlugin : public plugin::PluginBase {
+public:
+	OnboardComputerStatusPlugin() : PluginBase(),
+	status_nh("~onboard_computer")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		status_sub = status_nh.subscribe("status", 10, &OnboardComputerStatusPlugin::status_cb, this);
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return { /* Rx disabled */ };
+	}
+
+private:
+	ros::NodeHandle status_nh;
+	ros::Subscriber status_sub;
+
+	/**
+	 * @brief Send onboard computer status to FCU and groundstation
+	 *
+	 * Message specification: https://mavlink.io/en/messages/common.html#ONBOARD_COMPUTER_STATUS
+	 * @param req	received OnboardComputerStatus msg
+	 */
+	void status_cb(const mavros_msgs::OnboardComputerStatus::ConstPtr &req)
+	{
+		mavlink::common::msg::ONBOARD_COMPUTER_STATUS status {};
+
+		status.time_usec = req->time_usec;
+		status.uptime = req->uptime;
+		status.type = req->type;
+
+		const auto &cpu_cores = req->cpu_cores;
+		std::copy(cpu_cores.cbegin(), cpu_cores.cend(), status.cpu_cores.begin());
+
+		const auto &cpu_combined = req->cpu_combined;
+		std::copy(cpu_combined.cbegin(), cpu_combined.cend(), status.cpu_combined.begin());
+
+		const auto &gpu_cores = req->gpu_cores;
+		std::copy(gpu_cores.cbegin(), gpu_cores.cend(), status.gpu_cores.begin());
+
+		const auto &gpu_combined = req->gpu_combined;
+		std::copy(gpu_combined.cbegin(), gpu_combined.cend(), status.gpu_combined.begin());
+
+		status.temperature_board = req->temperature_board;
+
+		const auto &temperature_core = req->temperature_core;
+		std::copy(temperature_core.cbegin(), temperature_core.cend(), status.temperature_core.begin());
+
+		const auto &fan_speed = req->fan_speed;
+		std::copy(fan_speed.cbegin(), fan_speed.cend(), status.fan_speed.begin());
+
+		status.ram_usage = req->ram_usage;
+		status.ram_total = req->ram_total;
+
+		const auto &storage_type = req->storage_type;
+		std::copy(storage_type.cbegin(), storage_type.cend(), status.storage_type.begin());
+
+		const auto &storage_usage = req->storage_usage;
+		std::copy(storage_usage.cbegin(), storage_usage.cend(), status.storage_usage.begin());
+
+		const auto &storage_total = req->storage_total;
+		std::copy(storage_total.cbegin(), storage_total.cend(), status.storage_total.begin());
+
+		const auto &link_type = req->link_type;
+		std::copy(link_type.cbegin(), link_type.cend(), status.link_type.begin());
+
+		const auto &link_tx_rate = req->link_tx_rate;
+		std::copy(link_tx_rate.cbegin(), link_tx_rate.cend(), status.link_tx_rate.begin());
+
+		const auto &link_rx_rate = req->link_rx_rate;
+		std::copy(link_rx_rate.cbegin(), link_rx_rate.cend(), status.link_rx_rate.begin());
+
+		const auto &link_tx_max = req->link_tx_max;
+		std::copy(link_tx_max.cbegin(), link_tx_max.cend(), status.link_tx_max.begin());
+
+		const auto &link_rx_max = req->link_rx_max;
+		std::copy(link_rx_max.cbegin(), link_rx_max.cend(), status.link_rx_max.begin());
```

Comment:
Sometimes usage of `const auto &` improve readability, sometimes not.
Here it makes it harder to understand.

Also i'd like to see `cog` generator for field copy.

Reply:
@vooon Thanks for reviewing! I'm not sure I understand what you mean by `Also i'd like to see cog generator for field copy.`. Could you elaborate on where exactly you want `cog` and what benefit it would bring?

Reply:
https://github.com/mavlink/mavros/blob/master/mavros/src/plugins/home_position.cpp#L137 here for example cog used to minimise manual duplications. So less room for copy-paste error.
So instead of writing all that copy manually i prefer something like that:
```
// [[[cog:
// for field in [......]:
//     cog.outl(f"status.{ field } = req->{ field };")
// 
// for field in ["link_rx_max", ...]:
//     cog.outl(f"std::copy(req->{ field }.cbegin(), req->{ field }.cend(), status.{ field }.begin());")
// ]]]

Reply:
Thanks! I never used that before but once I got the the syntax, I agree it's quite convenient :)